File: ios/Feats/Feats/Models/Post.swift
import Foundation

struct Post: Codable, Identifiable, Equatable {
    let id: String
    let userId: String
    let activityTypeId: String
    let description: String?
    let createdAt: Date
    let updatedAt: Date
    let user: User?
    let activityType: ActivityType?
    let images: [PostImage]?
    let reactions: [Reaction]?
    let commentCount: Int?

    enum CodingKeys: String, CodingKey {
        case id, description, user, images, reactions
        case userId = "user_id"
        case activityTypeId = "activity_type_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case activityType = "activity_type"
        case commentCount = "comment_count"
    }

    static func == (lhs: Post, rhs: Post) -> Bool {
        let lhsReactionSignature = lhs.reactions?.map { "\($0.userId):\($0.reactionType.rawValue)" } ?? []
        let rhsReactionSignature = rhs.reactions?.map { "\($0.userId):\($0.reactionType.rawValue)" } ?? []
        let lhsImageSignature = lhs.images?.map(\.id) ?? []
        let rhsImageSignature = rhs.images?.map(\.id) ?? []

        return lhs.id == rhs.id &&
            lhs.description == rhs.description &&
            lhs.commentCount == rhs.commentCount &&
            lhsReactionSignature == rhsReactionSignature &&
            lhsImageSignature == rhsImageSignature
    }
}

struct PostImage: Codable, Identifiable, Equatable {
    let id: String
    let postId: String
    let displayOrder: Int
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case postId = "post_id"
        case displayOrder = "display_order"
        case createdAt = "created_at"
    }
}

struct CreatePostRequest: Codable {
    let activityTypeId: String
    let description: String?

    enum CodingKeys: String, CodingKey {
        case activityTypeId = "activity_type_id"
        case description
    }
}

struct UpdatePostRequest: Codable {
    let description: String?
}

File: backend/internal/models/post.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Post struct {
	ID             string         `gorm:"type:text;primaryKey" json:"id"`
	GroupID        string         `gorm:"type:text;not null;index" json:"group_id"`
	UserID         string         `gorm:"type:text;not null;index" json:"user_id"`
	ActivityTypeID string         `gorm:"type:text;not null" json:"activity_type_id"`
	Description    *string        `gorm:"type:text" json:"description,omitempty"`
	CreatedAt      time.Time      `gorm:"type:datetime;not null;index" json:"created_at"`
	UpdatedAt      time.Time      `gorm:"type:datetime;not null" json:"updated_at"`
	DeletedAt      gorm.DeletedAt `gorm:"type:datetime;index" json:"-"`

	// Computed fields (not stored in DB)
	CommentCount int `gorm:"-" json:"comment_count"`

	// Relationships
	Group        Group        `gorm:"foreignKey:GroupID" json:"-"`
	User         User         `gorm:"foreignKey:UserID" json:"user,omitempty"`
	ActivityType ActivityType `gorm:"foreignKey:ActivityTypeID" json:"activity_type,omitempty"`
	Images       []PostImage  `gorm:"foreignKey:PostID" json:"images,omitempty"`
	Reactions    []Reaction   `gorm:"foreignKey:PostID" json:"reactions,omitempty"`
	Comments     []Comment    `gorm:"foreignKey:PostID" json:"comments,omitempty"`
}

func (Post) TableName() string {
	return "posts"
}

type PostImage struct {
	ID           string    `gorm:"type:text;primaryKey" json:"id"`
	PostID       string    `gorm:"type:text;not null;index" json:"post_id"`
	FilePath     string    `gorm:"type:text;not null" json:"-"`
	DisplayOrder int       `gorm:"not null" json:"display_order"`
	CreatedAt    time.Time `gorm:"type:datetime;not null" json:"created_at"`

	// Relationships
	Post Post `gorm:"foreignKey:PostID" json:"-"`
}

func (PostImage) TableName() string {
	return "post_images"
}

// ImageURL returns the public URL for the image
func (p *PostImage) ImageURL(baseURL string) string {
	return baseURL + "/images/" + p.ID
}
